pkg/kapp/diff/change_set_with_versioned_rs.go

```diff
@@ -276,18 +292,10 @@ func newVersionedResources(rs []ctlres.Resource) versionedResources {
 	return result
 }
 
-func existingVersionedResources(rs []ctlres.Resource) versionedResources {
+func (d ChangeSetWithVersionedRs) existingVersionedResources(rs []ctlres.Resource) versionedResources {
 	var result versionedResources
 	for _, res := range rs {
-		// Expect that versioned resources should not be transient
-		// (Annotations may have been copied from versioned resources
-		// onto transient resources for non-versioning related purposes).
-		_, hasVersionedAnn := res.Annotations()[versionedResAnnKey]
-
-		versionedRs := VersionedResource{res: res}
-		_, version := versionedRs.BaseNameAndVersion()
-
-		if hasVersionedAnn && !res.Transient() && version != "" {
+		if d.versionedResourceName(res).IsExistingVersioned() {
```

Comment:
Something like this might help in reducing the noise?
```suggestion
		if (hasVersionedAnn && !res.Transient() && version != "") || d.stripEnabled(res) {
```